src/io.c

```diff
@@ -449,9 +463,15 @@ dispatch_io_create_with_path(dispatch_io_type_t type, const char *path,
 				break;
 			default:
 				if ((path_data->oflag & O_CREAT) &&
-						(*(path_data->path + path_data->pathlen - 1) != '/')) {
+						!_is_separator(*(path_data->path + path_data->pathlen - 1))) {
 					// Check parent directory
-					char *c = strrchr(path_data->path, '/');
+					char *c = NULL;
+					for (size_t i = path_data->pathlen; i-- > 0;) {
```

Comment:
Although this code is correct, I found it confusing at first because it appears that the start index should be path_data->pathlen - 1, but this is not the case because i-- decrements before each pass of the loop. I understand the reason for the - size_t is unsigned. Maybe it would be clearer to cast to ssize_t:

for (ssize_t i = (ssize_t)path_data->pathlen - 1; i >= 0; i--)

Reply:
I forgot that Dispatch declares `ssize_t` on Windows. (It normally isn't present.) That sounds better, thanks!